source/material_model/grain_size.cc

```diff
@@ -486,12 +486,33 @@ namespace aspect
       const SymmetricTensor<2,dim> shear_strain_rate = strain_rate - 1./dim * trace(strain_rate) * unit_symmetric_tensor<dim>();
```

Comment:
Do we even need this whole function? It doesn't seem like this is what's being used to fill out.viscosities. 

(And in case this function is being used somewhere, I would also use it in evaluate(), otherwise this is duplicating code).